runtime/drivers/druid/druidsqldriver/druid_api_sql_driver.go

```diff
@@ -419,6 +421,27 @@ func (chc *coordinatorHTTPCheck) IsHardFailure(ctx context.Context) (bool, error
 type DruidQueryContext struct {
 	SQLQueryID                 string `json:"sqlQueryId"`
 	EnableTimeBoundaryPlanning bool   `json:"enableTimeBoundaryPlanning"`
+	UseCache                   *bool  `json:"useCache,omitempty"`
+	PopulateCache              *bool  `json:"populateCache,omitempty"`
+}
+
+type QueryCfg struct {
+	UseCache      bool
+	PopulateCache bool
+}
+type ctxKey int
+
+var queryCfgKey ctxKey
+
+func WithQueryCfg(ctx context.Context, cfg *QueryCfg) context.Context {
+	return context.WithValue(ctx, queryCfgKey, cfg)
+}
```

Comment:
1. Should be `QueryConfig` instead of `QueryCfg` – abbreviations are generally used for variable names, but not for exported type names.
2. The context key type should be `struct{}` not `int`. Search the codebase for `ctxKey` for examples.
3. Newline missing between lines 431 and 432
4. These definitions don't really fit here in the file – it's between `DruidQueryContext` and `DruidParameter` which are both Druid JSON types used in `DruidRequest`

Reply:
1. ok
2. ok, I looked the go [doc](https://github.com/golang/go/blob/master/src/context/context.go#L145) but anyways changed.
3. ok
4. moved to the end before unexported definitions.